alchemy/src/destroy.ts

```diff
@@ -102,14 +113,35 @@ export async function destroy<Type extends string>(
       return;
     }
 
+    const pendingDeletions = await state.output[ResourceScope]
+      .get<PendingDeletions>("pendingDeletions")
+      .catch((e) => {
+        if (e instanceof RootScopeStateAttemptError) {
+          return [];
+        }
+        throw e;
+      });
+    const pendingDeletion = pendingDeletions?.find(
+      (deletion) => deletion.resource[ResourceID] === instance[ResourceID],
+    );
+
+    if (options?.replace == null && pendingDeletion !== undefined) {
+      await destroy(pendingDeletion.resource, {
+        replace: {
+          props: pendingDeletion.oldProps,
+          output: pendingDeletion.resource,
+        },
+      });
+    }
```

Comment:
What is this special handling for? Can destroy be generic and not specialized? Why do we need `options?.replace`?

Reply:
1. moving to the error based replacement means we can probably remove this actually.
2. not sure I understand, could you elaborate?
3. how else would we destroy the old resource if we don't have a reference to the old props and the old output?

Reply:
removed the check, but we still need to keep replace around

Reply:
The main question I have is why does `destroy` need to differentiate between destroying a replaced resource vs destroying an orphaned resource?